src/components/sortableGridView/SortableGridItemAnimationWrapper.tsx

```diff
@@ -0,0 +1,127 @@
+
+  useAnimatedReaction(() => itemsOrder.value[id], (newOrder) => {
+    const newPosition = getPositionByOrder(newOrder);
+    translateX.value = withTiming(newPosition.x, animationConfig);
+    translateY.value = withTiming(newPosition.y, animationConfig);
+  });
+
+  const onGestureEvent = useAnimatedGestureHandler<PanGestureHandlerGestureEvent, {x: number; y: number}>({
+    onStart: (_, context) => {
+      context.x = translateX.value;
+      context.y = translateY.value;
+      shouldScaleItem.value = true;
+      shouldFrontItem.value = true;
+    },
+    onActive: ({translationX, translationY}, context) => {
+      translateX.value = context.x + translationX;
+      translateY.value = context.y + translationY;
+
+      // Swapping items
+      const oldOrder = itemsOrder.value[id];
+      const newOrder = getOrderByPosition(translateX.value, translateY.value);
+      if (oldOrder !== newOrder) {
+        const itemIdToSwap = Object.keys(itemsOrder.value).find((itemId) => itemsOrder.value[itemId] === newOrder);
+        if (itemIdToSwap) {
+          const newItemsOrder = Object.assign({}, itemsOrder.value);
+          newItemsOrder[id] = newOrder;
+          newItemsOrder[itemIdToSwap] = oldOrder;
+          itemsOrder.value = newItemsOrder;
+        }
+      }
+
+      // Handle scrolling
+      const lowerBound = scrollY.value;
+      const upperBound = scrollY.value + screenHeight - itemSize * 1.5;
+      const maxScrollValue = contentHeight - screenHeight;
+      const leftToScroll = maxScrollValue - scrollY.value;
+      if (translateY.value < lowerBound) {
+        const diff = Math.min(lowerBound - translateY.value, lowerBound);
+        scrollY.value -= diff;
+        context.y -= diff;
+        translateY.value = context.y + translationY;
+        scrollTo(scrollViewRef, 0, scrollY.value, false);
+      }
+      if (translateY.value > upperBound) {
+        const diff = Math.min(translateY.value - upperBound, leftToScroll);
+        scrollY.value += diff;
+        context.y += diff;
+        translateY.value = context.y + translationY;
+        scrollTo(scrollViewRef, 0, scrollY.value, false);
+      }
+    },
+    onEnd: () => {
+      const destination = getPositionByOrder(itemsOrder.value[id]);
+      translateX.value = withTiming(destination.x, animationConfig, () => {
+        shouldFrontItem.value = false;
+      });
+      translateY.value = withTiming(destination.y, animationConfig);
+    },
+    onFinish: () => {
+      shouldScaleItem.value = false;
+    }
+  });
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const scale = withSpring(shouldScaleItem.value ? 1.1 : 1);
+    const zIndex = shouldFrontItem.value ? 100 : 0;
+    return {
+      width: itemSize,
+      height: itemSize,
+      zIndex,
+      transform: [
+        {translateX: translateX.value},
+        {translateY: translateY.value},
+        {scale}
+      ]
+    };
+  });
+  return (
+    <Animated.View style={[ABSOLUTE_ITEM, animatedStyle]}>
```

Comment:
Since we're using now gesture-handler v2, have you considered using its new API 
https://docs.swmansion.com/react-native-gesture-handler/docs/quickstart/quickstart

Reply:
I did consider it, but:

- there's not a lot of documentation about it just yet
- I like that in the older API I have context built in, no need for another sharedValue.
- The new API is overpowered to what I managed to do (which is simpler).

Tbh, the new API might be key to making this a truly usable component.
The main issue I think it has is scrolling vs. dragging UX wise.
I would love if this component could have logic such that items are draggable only after a long-press , but I just couldn't make it work - I think the new API can solve it (with some manual gesture handler), but as I said - there's not much documentation or examples so I couldn't make it work during a GW where I learned a lot from scratch.
If this is something that you could follow up on (uilib team) that'd be awesome.